tests/app/Imports/Wake/RecatImporterTest.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+namespace App\Imports\Wake;
+
+use App\BaseFunctionalTestCase;
+use App\Models\Aircraft\Aircraft;
+use App\Models\Aircraft\RecatCategory;
+use Illuminate\Console\OutputStyle;
+use Illuminate\Support\Collection;
+use InvalidArgumentException;
+use Mockery;
+
+class RecatImporterTest extends BaseFunctionalTestCase
+{
+    private $import;
+
+    public function setUp(): void
+    {
+        parent::setUp();
+        $this->import = new RecatImporter();
+        $this->import->withOutput(Mockery::spy(OutputStyle::class));
+    }
+
+    /**
+     * @dataProvider badDataProvider
+     */
```

Comment:
ooh what does this do?

Reply:
It essentially saves you having to write a test function for every data combination when the test is performing the same actions - it just calls the test functions
 multiple times with all the arguments listed in the provider function :)